### ISSUE TITLE
error: could not determine data type of parameter $1 when trying to query tree entity

### ISSUE THREAD

**Issue by nimame** [typeorm/typeorm#3527]
**Issue type:**

[x] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

I am trying to query an entity with a self-referencing tree structure :
```ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  BaseEntity,
  Tree,
  TreeChildren, TreeParent
} from "typeorm";

@Entity("category")
@Tree("materialized-path")
export class Category extends BaseEntity {

  @PrimaryGeneratedColumn("uuid")
  id: string;

  @Column({length: 100})
  name: string;

  @TreeChildren()
  children: Category[];

  @TreeParent()
  parent: Category;

}
```
I would like to get a filtered part of the table, either flattened or as a tree structure (ideally, I would like to be able to do both).

If I understand correctly, the `findTrees` method of `TreeRepository` always returns the whole table. So in order to be able to join other tables and add WHERE statements I tried something like this:

```ts
// ...
  const category = new Category();
  const query = getTreeRepository(Category)
    .createDescendantsQueryBuilder("c", "categoryClosure", category)
    .leftJoin("c.taxonomy", "t")
    .where(`t."userId" = :userId`, {userId: req.user.id});
  const categories: Category[] = await query.getMany();
// ...
```
Since, I was hoping that by providing a new (empty) instance of the `Category` class as the third parameter of `createDescendantsQueryBuilder`, I could get a result that includes all roots and their children that match the WHERE condition.

nb: From `src/repository/TreeRepository.ts` I figure the second argument (here: "categoryClosure") is ignored for materialized path entities?

However, this leads to the following SQL:
```sql
SELECT "c"."id" AS "c_id", "c"."name" AS "c_name", "c"."mpath" AS "c_mpath", "c"."parentId" AS "c_parentId", "c"."taxonomyId" AS "c_taxonomyId" FROM "category" "c" 
LEFT JOIN "taxonomy" "t" ON "t"."id"="c"."taxonomyId" 
WHERE t."userId" = $2 AND "t"."id" = $3 
-- PARAMETERS: [undefined,"381f9e0c-b88a-4edb-a2d2-88d948e9e57d","2e46634e-826c-4104-b722-60b9ae6eb60b"]
```

and the following error:

```
error: could not determine data type of parameter $1
```

When I provide a non-empty `category`, I get the same error even though the `category.id` turns up in the parameters array instead of `undefined` (but still not in the SQL statement).

I also tried simply using `createQueryBuilder` to query the table, but the `mpath` and `parentId` columns are never returned, even if specified explicitly using `addSelect`.

Am I missing something or is it not possible to do what I intend with TypeORM? 

Also, is implementing the `ltree` type from PostgreSQL on the road map (it is not yet implemented, if I'm not mistaken)?

Thank you.

**Comment by nimame** [comment]
#3481

**Comment by Kononnable** [comment]
@nimame  Why did you referenced another issue? You find this as duplicate or is it related somehow? Without a short description nobody knows.

**Comment by imnotjames** [comment]
I believe this to have been fixed by #7821
